src/hb-buffer-serialize.cc

```diff
@@ -196,6 +196,47 @@ _hb_buffer_serialize_glyphs_json (hb_buffer_t *buffer,
   return end - start;
 }
 
+static unsigned int
+_hb_buffer_serialize_unicode_json (hb_buffer_t *buffer,
+					unsigned int start,
+					unsigned int end,
+					char *buf,
+					unsigned int buf_size,
+					unsigned int *buf_consumed)
+{
+	hb_glyph_info_t *info = hb_buffer_get_glyph_infos (buffer, nullptr);
+
+	*buf_consumed = 0;
+	for (unsigned int i = start; i < end; i++)
+	{
+		char b[1024];
+		char *p = b;
+		if (i)
+			*p++ = ',';
+
+		*p++ = '{';
+
+		APPEND ("\"u\":");
+		p += hb_max (0, snprintf (p, ARRAY_LENGTH (b) - (p - b), "%u", info[i].codepoint));
```

Comment:
Does JSON accept hex integers? I think we should use those if doable.

Reply:
No, sadly not.